src/main/java/com/google/sps/data/HttpServletUtils.java

```diff
@@ -37,4 +54,80 @@ public static String getParameter(HttpServletRequest request, String name, Strin
     }
     return value;
   }
+
+  /** Returns a BlobKey for the uploaded file, or null if the user didn't upload a file. */
+  public static BlobKey getUploadedFileBlobKey(HttpServletRequest request, String formInput) {
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+    List<BlobKey> blobKeys = blobs.get(formInput);
+
+    // User submitted form without selecting a file (dev server)
+    if (blobKeys == null || blobKeys.isEmpty()) {
+      return null;
+    }
+
+    // Our form only contains a single file input, so get the first index.
+    BlobKey blobKey = blobKeys.get(0);
+
+    // User submitted form without selecting a file (live server)
+    BlobInfo blobInfo = new BlobInfoFactory().loadBlobInfo(blobKey);
+    if (blobInfo == null || blobInfo.getSize() == 0) {
+      blobstoreService.delete(blobKey);
```

Comment:
Nit: Seems strange that we have to delete it if it's not valid, ideally it should only be created if it's valid.

Reply:
Yeah, I know, it is weird, but that is how Blobstore works. 
When the user submits the form, the POST is handled by the Blobstore API. The API creates the blob from the file uploaded and store it in datastore, and then passes the rewritten request to the servlet that handles the form.
So, basically, we cannot know if it is valid until it is created and, therefore, stored.